src/scripts/main.js

```diff
@@ -1,3 +1,24 @@
 'use strict';
 
-// write your code here
+function cloneColumn(columnIndex) {
+  const headRows = table.querySelectorAll('thead tr');
+  const bodyRows = table.querySelectorAll('tbody tr');
+  const footRows = table.querySelectorAll('tfoot tr');
+
+  const insertClonedCell = (rows) => {
+    rows.forEach((row) => {
+      const cells = row.children;
+      const clonedCell = cells[columnIndex].cloneNode(true);
+
+      row.insertBefore(clonedCell, cells[cells.length - 1]);
+    });
+  };
+
+  insertClonedCell(headRows);
+  insertClonedCell(bodyRows);
+  insertClonedCell(footRows);
+}
+
+const table = document.querySelector('table');
+
+cloneColumn(table, 1);
```

Comment:
The `cloneColumn` function is called with two arguments, `table` and `1`. However, the function is defined to take only one argument, `columnIndex`. This is a misunderstanding of the function's parameters.